js/src/apbct-public--2--public.js

```diff
@@ -179,19 +179,87 @@ function ctGetPixelUrl() {
 }
 
 function ctSetHasScrolled() {
-	if( ! apbctLocalStorage.isSet('ct_has_scrolled') || ! apbctLocalStorage.get('ct_has_scrolled') ) {
+	if ( ! apbctLocalStorage.isSet('ct_has_scrolled') || ! apbctLocalStorage.get('ct_has_scrolled') ) {
 		ctSetCookie("ct_has_scrolled", 'true');
 		apbctLocalStorage.set('ct_has_scrolled', true);
 	}
 }
 
 function ctSetMouseMoved() {
-	if( ! apbctLocalStorage.isSet('ct_mouse_moved') || ! apbctLocalStorage.get('ct_mouse_moved') ) {
+	if ( ! apbctLocalStorage.isSet('ct_mouse_moved') || ! apbctLocalStorage.get('ct_mouse_moved') ) {
 		ctSetCookie("ct_mouse_moved", 'true');
 		apbctLocalStorage.set('ct_mouse_moved', true);
 	}
 }
 
+//init listeners for keyup and focus events
+function ctStartFieldsListening() {
+
+	if (
+		(apbctLocalStorage.isSet('ct_has_key_up') || apbctLocalStorage.get('ct_has_key_up'))
+		&&
+		(apbctLocalStorage.isSet('ct_has_input_focused') || apbctLocalStorage.get('ct_has_input_focused'))
+	) {
+		//already set
+		return
+	}
+
+	let forms = ctGetPageForms();
+	ctPublic.handled_fields = []
+
+	if (forms.length > 0) {
+		for (let i = 0; i < forms.length; i++) {
+			//handle only inputs and textareas
+			let handled_form_fields = forms[i].querySelectorAll('input,textarea')
+			for (let i = 0; i < handled_form_fields.length; i++) {
+				if (handled_form_fields[i].type !== 'hidden') {
+					//collect handled fields to remove handler in the future
+					ctPublic.handled_fields.push(handled_form_fields[i])
+					//do attach handlers
+					apbct_attach_event_handler(handled_form_fields[i], "focus", ctFunctionHasInputFocused)
+					apbct_attach_event_handler(handled_form_fields[i], "keyup", ctFunctionHasKeyUp)
+				}
+			}
+		}
+	}
+}
+
+//stop listening keyup and focus
+function ctStopFieldsListening(event_name, function_name) {
+	if (typeof ctPublic.handled_fields !== 'undefined' && ctPublic.handled_fields.length > 0) {
+		for (let i = 0; i < ctPublic.handled_fields.length; i++) {
+			apbct_remove_event_handler(ctPublic.handled_fields[i], "focus", function_name)
```

Comment:
Replace "focus" event to variable `event_name`

Reply:
Done, thanks!